deepcell/data/_auth.py

```diff
@@ -0,0 +1,95 @@
+"""User interface to authentication layer for data/models."""
+
+import os
+import requests
+from pathlib import Path
+from tqdm import tqdm
+import logging
+
+
+# TODO: Update when name is registered
+_api_endpoint = "http://52.8.155.84/api/getData/"
+
+
+# TODO s:
+#  - Add instructions about how to create an account/access token to the
+#    docstring
+#  - Finalize name (i.e. references to `deepcell-connect` -> whatever the URL
+#    ends up being, e.g. users.deepcell.org
+#  - Add data caching + force option
+#  - Make download location a kwarg?
+def fetch_data(asset_key: str) -> None:
+    """Fetch assets through deepcell-connect authentication system.
+
+    Download assets from the deepcell suite of datasets and models which
+    require user-authentication.
+
+    .. note::
+
+       You must have a Deepcell Access Token set as an environment variable
+       with the name ``DEEPCELL_ACCESS_TOKEN`` in order to access assets.
+
+    Args:
+        :param asset_key: Key of the file to download.
+        The list of available assets can be found on the deepcell-connect 
+        homepage.
+    """
+    logging.basicConfig(level=logging.INFO)
+
+    # Download location - TODO: do we instead want to 
+    download_location = Path.home() / ".deepcell"
+    download_location.mkdir(exist_ok=True)
+
+    # Check for access token
+    access_token = os.environ.get("DEEPCELL_ACCESS_TOKEN")
+    if access_token is None:
+        raise ValueError(
+            (
+                "DEEPCELL_ACCESS_TOKEN not found.\n",
+                "Please set your access token to the DEEPCELL_ACCESS_TOKEN\n",
+                "environment variable.\n",
+                "For example:\n\n",
+                "\texport DEEPCELL_ACCESS_TOKEN=<your-token>.\n\n",
+                "If you don't have a token, create one at deepcell-connect.",
+            )
+        )
+
+    # Extract the filename from the asset_key, which can be a full path
+    fname = os.path.split(asset_key)[-1]
+
+    # Request download URL
+    headers = {"X-Api-Key": access_token}
+    logging.info("Making request to server")
+    resp = requests.post(
+        _api_endpoint, headers=headers, data={"s3_key": asset_key}
+    )
+    resp.raise_for_status()  # Raise exception if not 200 status
```

Comment:
Another TODO: Might want to catch `404`'s specifically and raise with the message body which is much more informative than the default HTTPError message.